subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/worker/device/Serial.kt

```diff
@@ -0,0 +1,32 @@
+package com.avito.runner.service.worker.device
+
+import com.google.common.net.InetAddresses
+
+sealed class Serial(open val value: String) {
+
+    data class Remote(override val value: String) : Serial(value) {
+        override fun toString(): String {
+            return value
+        }
+    }
+
+    data class Local(override val value: String) : Serial(value) {
+        override fun toString(): String {
+            return value
+        }
+    }
+
+    companion object {
+        fun from(value: String): Serial {
+            return if (isRemote(value)) Remote(
+                value
+            ) else Local(value)
+        }
+
+        @Suppress("UnstableApiUsage")
+        private fun isRemote(serial: String): Boolean {
```

Comment:
Propose to add serial validation. We need to fail if serial will be unexpected

Reply:
Do you know these cases? 
I don't see what value might be broken or illegal.

Reply:
It's not clear for me what serials could pass there.
I want to have a comment or test where we describe possible serials explicitly.

Reply:
This is a serial number from adb.

https://developer.android.com/studio/command-line/adb#devicestatus

```
adb devices
List of devices attached
emulator-5556 device product:sdk_google_phone_x86_64 model:Android_SDK_built_for_x86_64 device:generic_x86_64
emulator-5554 device product:sdk_google_phone_x86 model:Android_SDK_built_for_x86 device:generic_x86
0a388e93      device usb:1-1 product:razor model:Nexus_7 device:flo
10.0.0.5:5555 device
```
We have no control and contract for this value. It's just a string or ip address.

Test: subprojects/gradle/runner/service/src/test/kotlin/com/avito/runner/service/worker/device/adb/AdbDeviceParserTest.kt